### ISSUE TITLE
CI check to make sure SPM checksum matches the code

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-cocoa#3861]
SPM has a checksum: https://github.com/getsentry/sentry-cocoa/blob/eb077dd952eb309235943f301ee6d2943c33e124/Package.swift#L16

On the release of https://github.com/getsentry/sentry-cocoa/releases/tag/8.22.3-beta.0 it turned out SPM installation failed due to this checksum not matching.

We need a check in CI that makes sure this is OK. So we make sure no releases go out that are guaranteed to fail to install.

**Comment by philprime** [comment]
@itaybre can this be handled with your current refactoring of the release artifact handling?

**Comment by philippsuess** [comment]
Yes, if we fix the chicken egg problem with the release artifact handling, we could validate the checksum almost for free.

**Comment by philprime** [comment]
@itaybre do you think we can close this now that we publish binaries to https://github.com/getsentry/sentry-apple-binaries?
